### ISSUE TITLE
Feature Request: ZoneControl:ContaminantController capable of accepting ZoneList objects

### ISSUE THREAD

**Issue by mldichter** [NREL/EnergyPlus#7664]
Changing the Zone Name field to accept ZoneList names as well would be greatly appreciated.

```
ZoneControl:ContaminantController,
    CO2 Controller1,         !- Name
    EAST ZONE,               !- Zone Name
    CO2AvailSchedule,        !- Carbon Dioxide Control Availability Schedule Name
    CO2SetpointSchedule,     !- Carbon Dioxide Setpoint Schedule Name
    ,                        !- Minimum Carbon Dioxide Concentration Schedule Name
    ,                        !- Maximum Carbon Dioxide Concentration Schedule Name
    ,                        !- Generic Contaminant Control Availability Schedule Name
    ;                        !- Generic Contaminant Setpoint Schedule Name
```

**Comment by jmarrec** [comment]
Similar issue to #7557 (`ZoneVentilation:WindandStackOpenArea` to accept ZoneList as well). Probably best addressed in conjunction.

**Comment by mldichter** [comment]
There are other objects, in addition to these two, like this. I'm willing to go through all the objects and their fields in the documentation to make a list, but is there already a list like that? The IDFs become so much more readable using ZoneLists. I would gladly devote some time to this.

**Comment by jmarrec** [comment]
@mldichter the IDD is probably the best way to compile the list easily and programmatically. A python script for eg would be easy to return objects that have fields with \reference ZoneNames but not \reference ZoneAndZoneListNames (references from memory so approximative...).

I can help with that part if you want? then you can go through that list to see which could actually be changed (perhaps some cannot)

**Comment by mldichter** [comment]
@jmarrec I went through the IDD file. I think most of the stuff would be useful, especially for something like a hotel, apartment complex, or 10 story office building. The HVACTemplate:* objects I found would be nice for the user to specify the equipment that's autosized or for identical zones once and have the object apply to a ZoneList.

The only one that's a little weird is the Refrigeration:Case object. Having a residential size refrigerator object with curves, use schedules, etc. would be nice in apartment complexes, similar to how the ElectricEquipment and Lights objects apply to multiple zones.

Here's what I got. I want to make a new Feature Request for all these objects, close the issues for ZoneControl:ContaminantController and ZoneVentilation:WindandStackOpenArea, as well as a request for a residential refrigerator object. What do you think?
RoomAirModelType
RoomAir:TemperaturePattern:UserDefined
RoomAirSettings:OneNodeDisplacementVentilation
RoomAirSettings:ThreeNodeDisplacementVentilation
RoomAirSettings:CrossVentilation
RoomAirSettings:UnderFloorAirDistributionInterior
RoomAirSettings:UnderFloorAirDistributionExterior
ElectricEquipment:ITE:AirCooled
ZoneBaseboard:OutdoorTemperatureControlled
ZoneContaminantSourceAndSink:CarbonDioxide
ZoneContaminantSourceAndSink:Generic:Constant
ZoneContaminantSourceAndSink:Generic:CutoffModel
ZoneContaminantSourceAndSink:Generic:DecaySource
ZoneContaminantSourceAndSink:Generic:DepositionRateSink
Output:IlluminanceMap
ZoneInfiltration:EffectiveLeakageArea
ZoneInfiltration:FlowCoefficient
ZoneVentilation:WindandStackOpenArea
ZoneAirBalance:OutdoorAir
ZoneEarthtube
ZoneCoolTower:Shower
AirflowNetwork:MultiZone:Zone

HVACTemplate:Zone:IdealLoadsAirSystem
HVACTemplate:Zone:BaseboardHeat
HVACTemplate:Zone:FanCoil
HVACTemplate:Zone:PTAC
HVACTemplate:Zone:PTHP
HVACTemplate:Zone:WaterToAirHeatPump
HVACTemplate:Zone:VRF
HVACTemplate:Zone:Unitary
HVACTemplate:Zone:VAV
HVACTemplate:Zone:VAV:FanPowered
HVACTemplate:Zone:VAV:HeatAndCool
HVACTemplate:Zone:ConstantVolume
HVACTemplate:Zone:DualDuct

ZoneControl:Humidistat
ZoneControl:ContaminantController

Refrigeration:Case

WaterUse:Equipment

HybridModel:Zone

**Comment by mldichter** [comment]
Closed this issue and opened #7687 to include the feature request.
